distributed/cli/tests/test_dask_worker.py

```diff
@@ -711,19 +664,46 @@ def test_error_during_startup(monkeypatch, nanny, loop):
     monkeypatch.setenv("DASK_SCHEDULER_ADDRESS", scheduler_addr)
     with popen(
         [
-            "dask-scheduler",
+            "dask",
+            "scheduler",
             f"--port={scheduler_port}",
             "--dashboard-address=:0",
         ],
     ):
         with Client(scheduler_addr, loop=loop) as c:
             with popen(
                 [
-                    "dask-worker",
+                    "dask",
+                    "worker",
                     scheduler_addr,
                     nanny,
                     # This should clash due to a port conflict
                     f"--worker-port={scheduler_port}",
                 ],
             ) as worker:
                 assert worker.wait(10) == 1
+
+
+def test_deprecated_single_executable(loop):
+    port = open_port()
+    scheduler_addr = f"tcp://127.0.0.1:{port}"
+    with popen(
+        [
+            "dask",
+            "scheduler",
+            "--no-dashboard",
+            f"--port={port}",
+        ],
+    ) as _:
+        with Client(f"127.0.0.1:{port}", loop=loop) as c:
+            with popen(
+                [
+                    "dask-worker",
+                    scheduler_addr,
+                    f"--worker-port={port}",
+                ],
+                capture_output=True,
+            ) as worker:
+                stdout, stderr = worker.communicate()
+                logs = stdout.decode()
+                assert "FutureWarning: dask-worker is deprecated" in logs
```

Comment:
Alright I've reworked this test to confirm that `c.run` works on a function that returns `1+1`, but I separated that part from the `"FutureWarning ..." in stdout.decode()` check because `capture_output=True` combined with an `await c.run(...)` call is timing out for me. Here's another place where I'm happy to hear ideas. I think this is hitting my current limit of async usage knowledge

Reply:
awesome! thanks for the recommendation